hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/unique/InformixUniqueDelegate.java

```diff
@@ -9,31 +9,42 @@
 import org.hibernate.boot.Metadata;
 import org.hibernate.boot.model.relational.SqlStringGenerationContext;
 import org.hibernate.dialect.Dialect;
-import org.hibernate.dialect.unique.AlterTableUniqueDelegate;
+import org.hibernate.dialect.unique.SkipNullableUniqueDelegate;
 import org.hibernate.mapping.UniqueKey;
 
 /**
- * Informix requires the constraint name to come last on the alter table.
+ * Informix requires the constraint name to come last.
  * 
  * @author Brett Meyer
  */
-public class InformixUniqueDelegate extends AlterTableUniqueDelegate {
+public class InformixUniqueDelegate extends SkipNullableUniqueDelegate {
 	
 	public InformixUniqueDelegate( Dialect dialect ) {
 		super( dialect );
 	}
 
-	// legacy model ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+	@Override
+	protected void appendUniqueConstraint(StringBuilder fragment, UniqueKey uniqueKey) {
+		if ( !uniqueKey.hasNullableColumn() ) {
+			fragment.append( ", " );
+			fragment.append( uniqueConstraintSql(uniqueKey) );
+			if ( uniqueKey.isNameExplicit() ) {
+				fragment.append( " constraint " ).append( uniqueKey.getName() );
+			}
+		}
+	}
 
 	@Override
 	public String getAlterTableToAddUniqueKeyCommand(UniqueKey uniqueKey, Metadata metadata,
 			SqlStringGenerationContext context) {
-		// Do this here, rather than allowing UniqueKey/Constraint to do it.
-		// We need full, simplified control over whether it happens.
-		final String tableName = context.format( uniqueKey.getTable().getQualifiedTableName() );
-		final String constraintName = dialect.quote( uniqueKey.getName() );
-		return dialect.getAlterTableString( tableName )
-				+ " add constraint " + uniqueConstraintSql( uniqueKey ) + " constraint " + constraintName;
+		if (uniqueKey.hasNullableColumn() || !context.isMigration()){
```

Comment:
Why do you need the `!context.isMigration()` condition?

Reply:
Since the parent has changed to SkipNullableUniqueDelegate, which is a descendant of CreateTableUniqueDelegate and I cannot call super.getAlterTableToAddUniqueKeyCommand(...), so I tried to replicate the parents logic.

Reply:
The implementation in `SkipNullableUniqueDelegate` is:

```java
	public String getAlterTableToAddUniqueKeyCommand(UniqueKey uniqueKey, Metadata metadata, SqlStringGenerationContext context) {
		return uniqueKey.hasNullableColumn() ? "" : super.getAlterTableToAddUniqueKeyCommand( uniqueKey, metadata, context );
	}
```

Reply:
Yes, but there's also super.getAlterTableToAddUniqueKeyCommand() where the implementation is
```
	@Override
	public String getAlterTableToAddUniqueKeyCommand(UniqueKey uniqueKey, Metadata metadata, SqlStringGenerationContext context) {
		return context.isMigration() ? super.getAlterTableToAddUniqueKeyCommand( uniqueKey, metadata, context ) : "";
	}

```

Reply:
Indeed, sorry for the noise.